internal/adminapi/endpoints.go

```diff
@@ -165,49 +159,36 @@ func adminAPIFromEndpoint(
 	// NOTE: Endpoint's addresses are assumed to be fungible, therefore we pick
 	// only the first one.
 	// For the context please see the `Endpoint.Addresses` godoc.
-	eAddress := endpoint.Addresses[0]
+	podIPAddr := endpoint.Addresses[0]
+	if addressFamily == discoveryv1.AddressTypeIPv6 {
+		podIPAddr = fmt.Sprintf("[%s]", podIPAddr)
+	}
 
 	// NOTE: We assume https below because the referenced Admin API
 	// server will live in another Pod/elsewhere so allowing http would
 	// not be considered best practice.
 
-	switch dnsStrategy {
-	case cfgtypes.ServiceScopedPodDNSStrategy:
-		if service.Name == "" {
-			return DiscoveredAdminAPI{}, fmt.Errorf(
-				"service name is empty for an endpoint with TargetRef %s/%s",
-				endpoint.TargetRef.Namespace, endpoint.TargetRef.Name,
-			)
-		}
-
-		ipAddr := strings.ReplaceAll(eAddress, ".", "-")
-		address := fmt.Sprintf("%s.%s.%s.svc", ipAddr, service.Name, service.Namespace)
-
-		return DiscoveredAdminAPI{
-			Address: fmt.Sprintf("https://%s:%d", address, *port.Port),
-			PodRef:  podNN,
-		}, nil
-
-	case cfgtypes.NamespaceScopedPodDNSStrategy:
-		ipAddr := strings.ReplaceAll(eAddress, ".", "-")
-		address := fmt.Sprintf("%s.%s.pod", ipAddr, service.Namespace)
-
-		return DiscoveredAdminAPI{
-			Address: fmt.Sprintf("https://%s:%d", address, *port.Port),
-			PodRef:  podNN,
-		}, nil
-
-	case cfgtypes.IPDNSStrategy:
-		bounded := eAddress
-		if addressFamily == discoveryv1.AddressTypeIPv6 {
-			bounded = fmt.Sprintf("[%s]", bounded)
-		}
-		return DiscoveredAdminAPI{
-			Address: fmt.Sprintf("https://%s:%d", bounded, *port.Port),
-			PodRef:  podNN,
-		}, nil
-
-	default:
-		return DiscoveredAdminAPI{}, fmt.Errorf("unknown dns strategy: %s", dnsStrategy)
+	if service.Name == "" {
+		return DiscoveredAdminAPI{}, fmt.Errorf(
+			"service name is empty for an endpoint with TargetRef %s/%s",
+			endpoint.TargetRef.Namespace, endpoint.TargetRef.Name,
+		)
 	}
+
+	return DiscoveredAdminAPI{
+		// Address format:
+		//   - ipv4 - https://10.244.0.16:8444
+		//   - ipv6 - https://[fd00:10:244::d]:8444
+		Address: fmt.Sprintf("https://%s:%d", podIPAddr, *port.Port),
+		// TLSServerName format:
+		//   - pod.ipv4 - pod.dataplane-admin-kong-rqwr9-sc49t.default.svc
+		// Currently everywhere (KGO, Chart) certificates are generated like that
+		// *.<service-name>.<namespace>.svc e.g.: *.dataplane-admin-kong-rqwr9-sc49t.default.svc
+		// so we are following the same pattern here, it could anything else than pod as the first
+		// part. Changing this would require changes in the certificate generation logic and may
+		// break users' setups. Not using something based on IP for the first part makes it work
+		// both for IPv4 and IPv6 without any special treatment.
```

Comment:
- Can we have a little more explanation here why we need the `pod` prefix here instead of the dash separate IP address?
- Any reason this is only covered for ipv4? I don't see a separate bullet point for ipv6 so I'd assume both work the same yet this is still a bull point list 🤔

Reply:
I reworded it to make it clear.

I use `pod` prefix to fulfill the wildcard certificate requirements that KGO/Chart generates. It can be any word, so let's use something simple instead of constructing something based on IPs, etc. when it's meaningless. 

I haven't decided to go with `<service-name>.<namespace>.svc`, because it will make KIC with this change incompatible with previous versions of KGO/Chart, so I decided to avoid this friction. I think when KO, becomes a thing it can be adjusted. I'll create an issue

Reply:
> I haven't decided to go with <service-name>.<namespace>.svc, because it will make KIC with this change incompatible with previous versions of KGO/Chart

Can you point where that's being generated this way (I'm just curious and also believe pointed to a concrete implementation would be beneficial).

Reply:
I documented this. An interesting finding is that for Chart we've never supported this properly (see the updated comment). Furthermore, I had to get rid of https://github.com/Kong/kubernetes-ingress-controller/commit/1a6f7ccdf51f8ff49b495c4c74f8fd6a748d48c6#diff-540e5a66571fe7ea83c6641fb9bc48b485ee07dce80a1f28de9ddccd2bf77d0c to make it boot with code from this PR (chart always fills all the options) and `CONTROLLER_KONG_ADMIN_TLS_SKIP_VERIFY=true`. I'll write down possible improvements in an issue dedicated to KO

Reply:
Nicely written 👍 Thanks for adding more thorough explanation.

Don't we want to cover removing the validation with an issue?

Reply:
Promised issues
- https://github.com/Kong/kubernetes-ingress-controller/issues/7064
- https://github.com/Kong/gateway-operator/issues/1109